### PR TITLE
ignore empty values when updating rows

### PR DESCRIPTION
**Pull request by spuniani**

### CODE REVIEW

File: src/main/java/com/sarality/db/modifier/IgnoreEmptyValueColumnUpdater.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import android.content.ContentValues;
+import android.text.TextUtils;
+
+import com.sarality.db.Column;
+import com.sarality.db.RecordModifier;
+import com.sarality.db.content.ContentValueWriter;
+import com.sarality.db.io.ColumnValueReader;
+
+/**
+ * When updating a row, ignore any updates to a column when the value is null or empty
```
Comment (abhideep) [inline src/main/java/com/sarality/db/modifier/IgnoreEmptyValueColumnUpdater.java]: Sorry, but how are we going to use this? If we initialize this table definition, how will we update the column when we perform the sync from the cloud down to the app?

**Comment by spuniani** [comment]
The idea is to ignore the value if it is empty - which would be true only
for local updates to the table. When getting the value from the cloud, the
global Id will be populated, so it will not be ignored.

**Comment by abhideep** [comment]
I was wondering how it will be applied. Makes sense now. Thanks.
This applies to both the GlobalId and GlobalVersion, right? If so, can we do one thing? Let's take a set/list of columns rather than one coulmn at a time and let's not call it Updater. EmptyColumnValueRemover maybe?

**Comment by abhideep** [comment]
Yes, we could do that, but might be better to keep the populator complete and change behaviour based on need. This way if someone wants to carry the globalId and version along with them all along they can do that as well. 

Either one works really, so your call.

**Pull request merged**